src/common/deconvolution_pd.hpp

```diff
@@ -169,6 +169,20 @@ struct deconvolution_pd_t : public primitive_desc_t {
         }
         return ok;
     }
+
+    bool expect_data_types(data_type_t src_dt, data_type_t wei_dt,
+            data_type_t bia_dt, data_type_t dst_dt) const {
+        bool ok = true
+                && (src_dt == data_type::undef
+                        || invariant_src_md()->data_type == src_dt)
+                && (wei_dt == data_type::undef
+                        || invariant_wei_md()->data_type == wei_dt)
+                && (dst_dt == data_type::undef
+                        || invariant_dst_md()->data_type == dst_dt);
+        if (with_bias() && bia_dt != data_type::undef)
+            ok = ok && invariant_bia_md()->data_type == bia_dt;
+        return ok;
+    }
```

Comment:
Hi! When do you anticipate data types coming to this function undefined?

Reply:
We are just following the oneDNN standard, like in convolution_pd.hpp [here](https://github.com/oneapi-src/oneDNN/blob/e26d1f66e7cc3c4f086b993d74023c7fc9459d4b/src/common/convolution_pd.hpp#L213C10-L213C10).

Reply:
I see, thanks. Could you, please, add a comment that the body follows `convolution_pd_t::expect_data_types(...)` function?